pkg/systemd/services/service-details.jsx

```diff
@@ -278,35 +293,87 @@ export class ServiceDetails extends React.Component {
         const args = [[this.props.unit.Names[0]], false];
         if (force !== undefined)
             args.push(force == "true");
-        systemd_client[this.props.owner].call(SD_OBJ, SD_MANAGER, method, args)
+        const promise = systemd_client[this.props.owner].call(SD_OBJ, SD_MANAGER, method, args)
                 .then(([results]) => {
                     if (results.length == 2 && !results[0])
                         this.show_note(_("This unit is not designed to be enabled explicitly."));
                     /* Executing daemon reload after file operations is necessary -
                      * see https://github.com/systemd/systemd/blob/main/src/systemctl/systemctl.c [enable_unit function]
                      */
                     systemd_client[this.props.owner].call(SD_OBJ, SD_MANAGER, "Reload", null);
-                })
-                .catch(error => {
-                    this.show_error(error.toString());
-                    this.setState({ waitsFileAction: false });
                 });
+        if (catchExc) {
+            return promise.catch(error => {
+                this.show_error(error.toString());
+                this.setState({ waitsFileAction: false });
+            });
+        } else {
+            return promise;
+        }
+    }
+
+    deleteAction() {
+        this.getUnitPaths().then(unitPaths => {
+            this.setState({ showDeleteDialog: true, unitPaths });
+        });
+    }
+
+    async getUnitPaths() {
+        const paths = [this.props.unit.FragmentPath];
+
+        await Promise.all(this.props.unit.Triggers.map(async trigger => {
+            // Getting dbus properties from a non-loaded unit is not possible so resort to systemctl show
+            const unitPath = await cockpit.spawn(["systemctl", "show", "--value",
+                "--property", "FragmentPath", trigger]);
+            paths.push(unitPath.trim());
+        })).catch(err => console.error("failed to look up unit details:", err.toString()));
+
+        return paths;
+    }
+
+    deleteTimer() {
+        // Stop timer so we don't get race conditions when the unit is gone.
+        if (this.interval) {
+            clearInterval(this.interval);
+            this.interval = null;
+        }
+
+        const promises = [];
+        if (this.props.unit.ActiveState === "active" || this.props.unit.ActiveState === "activating")
+            promises.push(this.unitAction("StopUnit", [], false));
+        if (this.props.unit.ActiveState === "failed")
+            promises.push(this.unitAction("ResetFailedUnit", [], false));
+        if (this.props.unit.UnitFileState === "enabled")
+            promises.push(this.unitFileAction("DisableUnitFiles", undefined, false));
+
+        promises.push(this.state.unitPaths.filter(path => path.startsWith("/etc/systemd/system"))
+                .map(path => cockpit.file(path, { superuser: "required" }).replace(null)));
+
+        return Promise.all(promises).then(() => {
```

Comment:
Ok, this does not fully work yet, as for some reason the promises are not executed in order while they should? So an error in unitAction still removes service/timer's.

Reply:
What do you mean with "in order"? The whole point of Promise.all() is to run all of them in parallel, and collect the results. If you need ordering, you need to run them as `first.then(() => second)`.

Reply:
I think I was confused about the order of output being the order of input which might not be execution order..